### PR TITLE
Search api added and documented

### PR DESCRIPTION
**Pull request by deepakhb2**

### CODE REVIEW

File: lib/flipkart_api.rb [hunk 1]
```diff
@@ -125,4 +125,29 @@ def get_product_by_id(product_id, format)
...
+
+  ##
+  #
+  #This method will get all the products accross those matches the key sent as parameters.
+  #Output will be json or xml depending on the parameter passed.
+  #Usage:
+  #  * fa.search("product name", "json", 5)
+  #
+  def searh(key, formate,  max_result=10)
+    rest_url = "#{@api}/search/format?query=#{key}&resultCount=#{max_result}"
```
Comment (technopreneurG) [inline lib/flipkart_api.rb:137]: I believe these 2 changes are needed:
- formate -> format
- "#{@api}/search/format?query - > "#{@api}/search/#{format}?query

File: flipkart_api.gemspec [hunk 2]
```diff
@@ -1,10 +1,10 @@
 Gem::Specification.new do |s|  
   s.name        = 'flipkart_api'
-  s.version     = '0.0.1'
+  s.version     = '0.0.2'
   s.date        = '2015-02-06'
```
Comment (technopreneurG) [inline flipkart_api.gemspec:4]: Hi, Should't this date get updated?

File: flipkart_api.gemspec [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
   s.summary     = "This gem is to pull data from flipkart using flipkart api"
   s.description = "Pull data from flipkart by using api"
-  s.authors     = ["Deepak HB"]
+  s.authors     = ["Deepak HB", "Girish Nair"]
```
Comment (technopreneurG) [inline flipkart_api.gemspec:7]: Aha! thanks :+1:

File: flipkart_api.gemspec (hunk 2 above)
Comment (deepakhb2) [inline flipkart_api.gemspec:4]: Yes... I missed it.. my mistake.. thanks..

**Pull request merged**